src/PhotometryTransform.cc

```diff
@@ -126,6 +126,63 @@ double integrateTn(int n) {
 }
 }  // namespace
 
+double PhotometryTransformChebyshev::oneIntegral(double x, double y) const {
+    lsst::geom::Point2D point = _toChebyshevRange(lsst::geom::Point2D(x, y));
+
+    // The nth/mth x/y chebyshev polynomial
+    // We need to compute through order+1, because the integral recurrance relation uses Tn+1(x)
+    Eigen::VectorXd Tnx(_order + 2);
+    Eigen::VectorXd Tmy(_order + 2);
+    // Algorithm: compute all the individual components recursively (since we'll need them anyway),
+    // then combine them into the final answer.
+    Tnx[0] = 1;
+    Tmy[0] = 1;
+    Tnx[1] = point.getX();
+    Tmy[1] = point.getY();
+    for (ndarray::Size i = 2; i <= _order + 1; ++i) {
+        Tnx[i] = 2 * point.getX() * Tnx[i - 1] - Tnx[i - 2];
+        Tmy[i] = 2 * point.getY() * Tmy[i - 1] - Tmy[i - 2];
+    }
+
+    // The integral of the nth/mth x/y chebyshev polynomial
+    Eigen::VectorXd integralTnx(_order + 1);
+    Eigen::VectorXd integralTmy(_order + 1);
+    integralTnx[0] = point.getX();
+    integralTmy[0] = point.getY();
+    if (_order > 0) {
+        integralTnx[1] = 0.5 * point.getX() * point.getX();
+        integralTmy[1] = 0.5 * point.getY() * point.getY();
+    }
+    for (ndarray::Size i = 2; i <= _order; ++i) {
+        // recurrence relation: integral(Tn(x)) = n*T[n+1](x)/(n^2 - 1) - x*T[n](x)/(n-1)
+        integralTmy[i] = i * Tmy[i + 1] / (i * i - 1) - point.getY() * Tmy[i] / (i - 1);
+        integralTnx[i] = i * Tnx[i + 1] / (i * i - 1) - point.getX() * Tnx[i] / (i - 1);
+    }
```

Comment:
Everything above is duplicated for x and y.  Can you separate that out into a 1-d function that's called twice?

Incidentally, this is also the part that I think would fit will in `lsst::geom::polynomials`; like the polynomials themselves, the integrals can be evaluated via:
 - a value for the zeroth term (i.e. line 150, above)
 - a value for the first term (i.e. line 153)
 - an expression for how to generate the nth term from the (n-1)th and (n-2)th terms (i.e. line 158).

So there, you'd implement a `Recurrence` object with just those definitions for a Chebyshev definite integral, and then you could use other machinery to manipulate them (i.e. loop to evaluate, combine x- and y- integrals to make a 2-d integral).  But all that said, I don't think that's worth doing now, given that you've got it working here; I'd want us to think about how to make the APIs there relate a polynomial to its integral (and/or derivatives), and right now there's no precedent for that and hence it's qualitatively new design work, which is always slow.

Reply:
Ok. At some point, we can lift this into `lsst::geom::polynomials`, and refactor this code to use that. I think the tests in here are pretty comprehensive, so they'll help that conversion.